### PR TITLE
Avoid merge small space with a far larger one

### PR DESCRIPTION
**Pull request by JiayuZzz**
### What is changed and how it works?

In space merge of PMEMAllocator, avoid merge a small space with a far larger one to save CPU overhead

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- CPU overhead of background merge thread drop from 100% to 2% in  the case that frequently do list pop (e.g. sequentially free space)

### CODE REVIEW

File: engine/pmem_allocator/free_list.cpp [hunk 1]
```diff
@@ -15,118 +15,176 @@ const uint32_t kMinMovableEntries = 1024;
...
+      last_lock = cur_lock;
     }
+    kvdk_assert(map_[cur].Empty(), "");
+    map_[cur] = Token(false, to_set);
+    cur += to_set;
+    remaining -= to_set;
   }
+// /*
+#if KVDK_DEBUG_LEVEL > 0
+  uint64_t debug_cur = offset + size;
```
Comment (ZiyanShi) [inline engine/pmem_allocator/free_list.cpp:50]: What's this code segment for? Please add comments or remove this.

File: engine/pmem_allocator/free_list.cpp [hunk 2]
```diff
@@ -15,118 +15,176 @@ const uint32_t kMinMovableEntries = 1024;
...
-  return merged;
+
+  return merged_size;
 }
 
 void Freelist::OrganizeFreeSpace() {
   MoveCachedEntriesToPool();
-  if (last_freed_after_merge_.load() > kMergeThreshold) {
+  if (false && last_freed_after_merge_.load() > kMergeThreshold) {
     MergeSpaceInPool();
```
Comment (ZiyanShi) [inline engine/pmem_allocator/free_list.cpp:188]: if (false) ?

**Review by ZiyanShi — APPROVED** [review]

File: engine/pmem_allocator/free_list.cpp (hunk 2 above)
Comment (JiayuZzz) [reply engine/pmem_allocator/free_list.cpp]: for test, resolved

**Pull request merged**